### PR TITLE
chore(Jenkinsfile): ensure that Docker ARM64 Linux are published

### PR DESCRIPTION
**Pull request by smerle33**
Ref. #477 

as deploy is outside the closure withPackerNode() the arm image is never published.

### CODE REVIEW

File: Jenkinsfile_k8s [hunk 1]
```diff
@@ -242,31 +242,20 @@ pipeline {
...
-                  if (${compute_type} == 'docker' && ${TAG_NAME} != null) {
-                    echo "Pushing jenkinsciinfra/jenkins-agent-${agent_type}:${TAG_NAME} & jenkinsciinfra/jenkins-agent-${agent_type}:latest"
-                    infra.withDockerPushCredentials {
+                  // if docker and building a tag, push to dockerhub from inside the node
+                  // else we would loose the docker image
+                  if (compute_type == 'docker' && tag_name != null) {
+                    stage('Publish Docker image') {
+                      echo "Pushing jenkinsciinfra/jenkins-agent-${agent_type}:${tag_name} & jenkinsciinfra/jenkins-agent-${agent_type}:latest"
+                      infra.withDockerPushCredentials {
                       sh 'docker push --all-tags jenkinsciinfra/jenkins-agent-${agent_type}'
```
Comment (lemeurherve) [inline Jenkinsfile_k8s:251]:
```suggestion
                        sh 'docker push --all-tags jenkinsciinfra/jenkins-agent-${agent_type}'
```

File: Jenkinsfile_k8s [hunk 2]
```diff
@@ -242,31 +242,20 @@ pipeline {
...
                     sh 'packer build -timestamp-ui -force -only="${PKR_VAR_image_type}.${PKR_VAR_agent_os_type}" ./'
                   }
-                  // si docker et buildtag alors :
-                  if (${compute_type} == 'docker' && ${TAG_NAME} != null) {
-                    echo "Pushing jenkinsciinfra/jenkins-agent-${agent_type}:${TAG_NAME} & jenkinsciinfra/jenkins-agent-${agent_type}:latest"
-                    infra.withDockerPushCredentials {
+                  // if docker and building a tag, push to dockerhub from inside the node
+                  // else we would loose the docker image
+                  if (compute_type == 'docker' && tag_name != null) {
+                    stage('Publish Docker image') {
```
Comment (lemeurherve) [inline Jenkinsfile_k8s:248]:
```suggestion
                    stage('Publish Docker images') {
```

Comment (dduportal) [reply Jenkinsfile_k8s:249]: The singular maps to the fact that there is only 1 image pushed.
Tags are child concepts of a given image (and platform are grand-child).


Suggestions (but not mandatory and not strong):
- `stage("Publish Docker image ${agent_type} for ${pkr_var_architecture}")
- `stage('Publish all tags for Docker image')`

**Comment by dduportal** [comment]
WDYT about the following PR title:

```
chore(Jenkinsfile): ensure that Docker ARM64 Linux are published
```

under the following arguments:

- This PR does not change anything in the packer images (VM and Docker) so not a `feat` or a `fix`
- It's a pipeline related change, so usually a `chore`
- The new title reflect the intent (what was done on the code is implementation detail)

**Review by dduportal — APPROVED** [review]
🚀

**Pull request merged**